mslice/models/slice/matplotlib_slice_plotter.py

```diff
@@ -20,200 +21,118 @@ def __init__(self):
+        comment = get_comment(str(workspace.name))
+        ax.get_xaxis().set_units(x_axis.units)
+        ax.get_yaxis().set_units(y_axis.units)
         # labels
-        plot_axes.set_xlabel(get_display_name(x_axis.units, comment), picker=PICKER_TOL_PTS)
-        plot_axes.set_ylabel(get_display_name(y_axis.units, comment), picker=PICKER_TOL_PTS)
-        plot_axes.set_xlim(x_axis.start)
-        plot_axes.set_ylim(y_axis.start)
-        plot_axes.get_xaxis().set_units(x_axis.units)
-        plot_axes.get_yaxis().set_units(y_axis.units)
-
-        # colorbar - have we plotted one previously?
-        try:
-            cb_axes = plt.gcf().get_axes()[1]
-        except IndexError:
-            cb_axes = None
-        if cb_axes is None:
-            cb = plt.colorbar(image, ax=plot_axes)
-        else:
-            cb = plt.colorbar(image, cax=cb_axes)
+        ax.set_xlabel(get_display_name(x_axis.units, comment), picker=PICKER_TOL_PTS)
+        ax.set_ylabel(get_display_name(y_axis.units, comment), picker=PICKER_TOL_PTS)
+        ax.set_xlim(x_axis.start)
+        ax.set_ylim(y_axis.start)
+        cb = plt.colorbar(image, ax=ax)
         cb.set_label('Intensity (arb. units)', labelpad=20, rotation=270, picker=PICKER_TOL_PTS)
-        plt.gcf().canvas.draw_idle()
-        plt.show()
-
-    def show_scattering_function(self, workspace):
-        cached_slice = self.slice_cache[workspace]
-        self._show_plot(workspace, cached_slice['plot_data'][0], cached_slice['boundaries'], cached_slice['colourmap'],
-                        cached_slice['norm'], cached_slice['momentum_axis'], cached_slice['energy_axis'])
-
-    def show_dynamical_susceptibility(self, workspace):
-        cached_slice = self.slice_cache[workspace]
-        if cached_slice['plot_data'][1] is None:
-            self.compute_chi(workspace)
-        self._show_plot(workspace, cached_slice['plot_data'][1], cached_slice['boundaries'], cached_slice['colourmap'],
-                        cached_slice['norm'], cached_slice['momentum_axis'], cached_slice['energy_axis'])
-
-    def show_dynamical_susceptibility_magnetic(self, workspace):
-        cached_slice = self.slice_cache[workspace]
-        if cached_slice['plot_data'][2] is None:
-            self.compute_chi_magnetic(workspace)
-        self._show_plot(workspace, cached_slice['plot_data'][2], cached_slice['boundaries'], cached_slice['colourmap'],
-                        cached_slice['norm'], cached_slice['momentum_axis'], cached_slice['energy_axis'])
+        cur_fig.canvas.draw_idle()
+        cur_fig.show()
+
+    def show_scattering_function(self, workspace_name):
+        slice_cache = self.slice_cache[workspace_name]
+        self._show_plot(slice_cache, slice_cache.scattering_function)
+
+    def show_dynamical_susceptibility(self, workspace_name):
+        slice_cache = self.slice_cache[workspace_name]
+        self._show_plot(slice_cache, slice_cache.chi)
```

Comment:
As per one of my testing comments, you could add a slightly different axis label for each of these cases as otherwise there is no visual difference between them on the plot